ext/rugged/rugged_object.c

```diff
@@ -346,14 +393,14 @@ static VALUE rb_git_object_type_GET(VALUE self)
 static VALUE rb_git_object_read_raw(VALUE self)
 {
 	git_object *object;
-	Data_Get_Struct(self, git_object, object);
+	TypedData_Get_Struct(self, git_object, &rugged_object_type, object);
 
 	return rugged_raw_read(git_object_owner(object), git_object_id(object));
 }
 
 void Init_rugged_object(void)
 {
-	rb_cRuggedObject = rb_define_class_under(rb_mRugged, "Object", rb_cObject);
+	rb_cRuggedObject = rb_define_class_under(rb_mRugged, "Object", rb_cData);
```

Comment:
I see you've read the docs. :smile: It seems like inheriting `rb_cData` is a requirement, but could this possibly break code in the Ruby part of the library? :thinking:

Reply:
Based on

    object.c:4398:    rb_cData = rb_define_class("Data", rb_cObject);

it should be fine since we end up with the base object anyway. The specific "git object" classes already have a level of indirection between them and base ruby object, so I don't expect issues. We would still however roll this out somewhat slowly and I would expect both this test suite and our applications' test suites to notice if it breaks.